PWG/EMCAL/EMCALbase/AliAnalysisTaskEmcalEmbeddingHelper.cxx

```diff
@@ -1602,9 +1689,27 @@ bool AliAnalysisTaskEmcalEmbeddingHelper::PythiaInfoFromCrossSectionFile(std::st
 void AliAnalysisTaskEmcalEmbeddingHelper::UserExec(Option_t*)
 {
   if (!fInitializedEmbedding) {
-    AliError("Chain not initialized before running! Setting up now.");
+    if ( fEmbeddedRunblock.size() == 0 ) {
+      AliError("Chain not initialized before running! Setting up now.");
+    }
     SetupEmbedding();
   }
+  
+  if ( fEmbeddedRunblock.size() > 0 ) {
+    // I do not think this is enough, I am not sure how to know if the train 
+    // was setup for run by run or run mixed analysis
+    Int_t runNumber = -1;
+    if ( AliAnalysisTaskSE::InputEvent() ) {
+      runNumber = AliAnalysisTaskSE::InputEvent()->GetRunNumber();
+    }
+    
+    if ( fDataRunNumber > -1 && fDataRunNumber != runNumber ) {
+      AliError(Form("CAREFUL! Check what you are doing, you are embedding a data run %d"
+                    " but block range was anchored to %d"
+                    " setup the train to do the analysis per Run",
+                    AliAnalysisTaskSE::InputEvent()->GetRunNumber(), fDataRunNumber));
+    }
+  }
```

Comment:
I think this condition could be refined: it's fine if it changes runs, it just can't go outside the runblock. I didn't totally understand your conditions when we first discussed this. However, I think it's probably fine for now since you'll run it run-by-run

Reply:
OK thanks